docs/DafnyRef/integration- rust/IntegrationRust.md

```diff
@@ -41,72 +41,139 @@ Hence, whenever a variable with move semantics (borrowed or owned) is used in a
-|                               | or optimized to `u8`..`i128`|
-| `(T1, T2...)`                 | `(T1, T2...)`               |
-| `seq<T>`                      | `Sequence<T: DafnyType>` (*)|
-| `set<T>`, `iset<T>`           | `Set<T: DafnyTypeEq>` (*)      |
-| `multiset<T>`                 | `MultiSet<T: DafnyTypeEq>` (*) |
-| `map<T,U>`, `imap<T,U>`       | `Map<T: DafnyTypeEq, U: DafnyTypeEq>` (*) |
-| function (arrow) types        | `Rc<dyn Fn(...) -> ...>`    |
-
-## With the option --raw-pointers=false (default)
-This version compiles classes to reference counting.
-However, it requires the nightly version to compile until the following feature
-is declared stable:
-- [trait Unsize](https://doc.rust-lang.org/book/appendix-07-nightly-rust.html)
-
-To install and use the nightly, use 
+- Class types are encoded in reference-counted pointers by default, but there is a command-line flag `--raw-pointers` that encodes class type into an equivalent of a raw pointer. The Dafny team will have to make a deallocation statement available in the future, as it is not available yet. Please weight in on [this issue](https://github.com/dafny-lang/dafny/issues/5257) if you are interested.
+
+For each construct, the third column indicates if it's in the Dafny runtime, is native built-in in Rust, or is a Rust crate.
+If runtime, it also indicates in parentheses what it is roughly equivalent to. `[[U]]` notation indicates the Rust equivalent of the Dafny type `U`.
+
+|  Dafny type                   |   Rust type                 | Defined in       |
+|-------------------------------|-----------------------------|------------------|
+| `bool`                        | `bool`                      | Native           |
+| `int`                         | `DafnyInt`                  | Runtime (`Rc<num::BigInt>`) |
+| `int64`                       | `i64`                       | Native           |
+| `int32`                       | `i32`                       | Native           |
+| `int16`                       | `i16`                       | Native           |
+| `int8`                        | `i8`                        | Native           |
+| `char`                        | `DafnyChar`                 | Runtime (`char`) |
+| `char` (`--unicode-chars=false`)| `DafnyCharUTF16`          | Runtime (`u16`)  |
+| `bv8` ... `bv128`             | `u8` ... `u128`             | Native           |
+| `type T = u: U | <range>`     | `[[U]]`                     | Native           |
+| `newtype T = u: U | range`    | `struct T([[U]])`           | Native           |
+| `newtype T = u: int | <range>`| `u8` ... `i128`             | Native           |
+| `ORDINAL`                     | N/A                         | TODO             |
+| `real`                        | Partially supported         | TODO             |
+| `seq<T>`                      | `Sequence<[[T]]>           `| Runtime (`enum`, either Array or Concat) |
```

Comment:
Thanks for adding the column, definitely way clearer! I would also be explicit that the exact runtime types are subject to change, and code that is not Dafny-generated should not rely on those details.

"either Array or Concat" especially is more detail than we should expose. You could imagine wanting to expand the enum in order to optimize sequences further in the future, and we don't want any one depending on Array and Concat being the only variants.